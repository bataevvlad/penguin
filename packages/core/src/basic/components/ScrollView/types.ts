import {
  ScrollViewProps as DefaultScrollViewProps,
  ScrollView as DefaultScrollView,
} from 'react-native';

export type IScrollViewProps = DefaultScrollViewProps;
export type IScrollView = DefaultScrollView;
