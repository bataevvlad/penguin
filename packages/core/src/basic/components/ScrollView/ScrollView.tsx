import {forwardRef} from 'react';
import {ScrollView as DefaultScrollView} from 'react-native';

import {IScrollView, IScrollViewProps} from './types';

// eslint-disable-next-line react/display-name
const ScrollView = forwardRef<IScrollView, IScrollViewProps>((props, ref) => {
  return <DefaultScrollView ref={ref} {...props} />;
});
export default ScrollView;
