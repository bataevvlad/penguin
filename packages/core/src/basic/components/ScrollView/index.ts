import ScrollView from './ScrollView';

export * from './types';

export {ScrollView};
