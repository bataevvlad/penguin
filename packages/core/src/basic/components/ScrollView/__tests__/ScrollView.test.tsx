import {render} from '@testing-library/react-native';

import ScrollView from '../ScrollView';

describe('@penguin/core/ScrollView', () => {
  it('renders correctly', () => {
    const {toJSON} = render(<ScrollView />);
    expect(toJSON()).toMatchSnapshot();
  });
});
