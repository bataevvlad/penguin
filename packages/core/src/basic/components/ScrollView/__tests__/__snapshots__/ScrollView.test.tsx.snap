// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@penguin/core/ScrollView renders correctly 1`] = `
<RCTScrollView>
  <View />
</RCTScrollView>
`;
